presto-geospatial-toolkit/src/main/java/com/facebook/presto/geospatial/serde/JtsGeometrySerde.java

```diff
@@ -60,10 +60,20 @@ public static Geometry deserialize(Slice shape)
             return readGeometry(input, type);
         }
         catch (TopologyException e) {
-            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, e.getMessage(), e);
+            if (safe) {
+                throw new PrestoException(INVALID_FUNCTION_ARGUMENT, e.getMessage(), e);
```

Comment:
Why the two variations of this method are needed?

In a place where `TopologyException` is expected the `PrestoException` can be caught, and then the `e.getCause()` can get examined.